django_saml2_auth/views.py

```diff
@@ -73,8 +73,9 @@ def acs(request: HttpRequest):
 
     # If RelayState params is passed, it is a JWT token that identifies the user trying to login
     # via sp_initiated_login endpoint
+    # RelayState, depending on the SAML app configuration, can also be a "/" (%2F char)
     relay_state = request.POST.get("RelayState")
-    if relay_state:
+    if relay_state and relay_state != '/':
```

Comment:
I think this must be adequately validated. Just checking if it exists or is not equal to something is not proper validation. WDYT?

Reply:
What about something like this:

```python
def is_jwt_well_formed(jwt: str):
    """Check if JWT is well formed

    Args:
        jwt (str): Json Web Token

    Returns:
        Boolean: True is JWT is well formed, otherwise False
    """
    if isinstance(jwt, str):
        # JWT should contain three segments, separated by two period ('.') characters.
        jwt_segments = jwt.split('.')
        if len(jwt_segments) == 3:
            jose_header = jwt_segments[0]
            # base64-encoded string length should be a multiple of 4
            if len(jose_header) % 4 == 0:
                try:
                    jh_decoded = base64.b64decode(jose_header).decode('utf-8')
                    if jh_decoded and jh_decoded.find('JWT') > -1:
                        return True
                except Exception:
                    return False
    # If tests not passed return False
    return False

token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiI0MjQyIiwibmFtZSI6Ikplc3NpY2EgVGVtcG9yYWwiLCJuaWNrbmFtZSI6Ikplc3MifQ.EDkUUxaM439gWLsQ8a8mJWIvQtgZe0et3O3z4Fd_J8o'
res = is_jwt_well_formed(jwt=token)
print(res)  # True
res = is_jwt_well_formed(jwt='/')
print(res)  # False
```

and then in views
```python
relay_state_is_token = is_jwt_well_formed(relay_state)
    if relay_state_is_token:
```

Reply:
There's no need to do this manually, as it might be error-prone. Decoding a JWT token actually validates it, too, in addition to checking all the fields, as documented [here](https://pyjwt.readthedocs.io/en/stable/usage.html#registered-claim-names).

Reply:
The idea was not to check the token manually, but to check wether the relay state could be a token to check.
The RelayState is not necessary a token (as stated [here](https://developer.okta.com/docs/concepts/saml/#single-idp-vs-multiple-idps) and [there](https://stackoverflow.com/a/34351756)), but can be whatever.
So it seemed better to check briefly wether to enter the block where it actually checks if the token is valid, otherwise we would not really know if the token is not encoded correctly or is simply not a token to be checked.

Reply:
Fair enough!